templates/CRM/Report/Form/Tabs/Instance.tpl

```diff
@@ -136,10 +136,17 @@
   CRM.$(function($) {
     var confirmed = false,
       formName = {/literal}"{$form.formName}"{literal};
-    $('#_qf_' + formName + '_submit_next, #_qf_' + formName + '_submit_save').click(function() {
+    $('#_qf_' + formName + '_submit_next, #_qf_' + formName + '_submit_save').click(function(e) {
       if ($('#is_navigation').prop('checked') && $('#parent_id').val() == '') {
-        var confirmMsg = {/literal}'{ts escape="js"}You have chosen to include this report in the Navigation Menu without selecting a Parent Menu item from the dropdown. This will add the report to the top level menu bar. Are you sure you want to continue?{/ts}'{literal}
-        return confirm(confirmMsg);
+        var confirmMsg = ts('You have chosen to include this report in the Navigation Menu without selecting a Parent Menu item from the dropdown. This will add the report to the top level menu bar. Are you sure you want to continue?');
+        CRM.confirm({
+          title: ts('Save to top level?'),
+          message: confirmMsg
+        })
+        .on('crmConfirm:no', function(e) {
```

Comment:
The reason the original code works is because window.confirm is a blocking event - it freezes the browser until the user clicks on it.
CRM.confirm is a non-blocking event, so to use it to block a form submission you'd have to do the opposite - cancel the event _first_ then call CRM.confirm, then trigger the event again on crmConfirm:yes. To avoid an infinite loop in that last step you could pass some extra param to $.trigger and pay attention to that param when listening to the click event to see whether the click should be blocked or not.